05-cluster-prerequisites.md

```diff
@@ -35,14 +35,25 @@ Following the steps below will result in the provisioning of the shared Azure re
+  >
+  > :book: The app team is starting to lay down the groundwork for a high availability, and they know that centralizing Azure resources might introduce single point of failures. Therefere, the app team is tasked to assess how the resources could be shared efficiently without losing reliablity. This is turn for the Azure Container Registry with the additional problematic that container images can be heavy artifacts to deal with, so they are considering having presence in multiple regions as well. This requirement comes in hand with distributing the exact same container images as this is for tween clusters that happens to live in different regions. Although managing registry instances per region individually will mitigate the risk of having a region down, this approach won’t be falling back automatically nor replicating images by requiring in both cases manual intervention and/or additional procedures.  That is the reason why, the team is now up for the `Premium` tier that offers [Geo-Replication](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-geo-replication) as a built-in feature giving them the ability to share a single registry, more availability, and at same time reduce the network latency.  The app team plan to geo replicate the registries to the same regions where their AKS clusters are going to be deployed (`East US 2` and `Central US`), as this is the recommendation to optimize the DNS resolution. They pay close attention to the replicated regions, as they want no more regions than needed since the business unit (bu0001) incurs `Premium` SKU registry fees for each region they desire to geo-replicate. Under this configuration, they will pay *two* times `Premium` per month to get region proximity to ensure no extra network egress fees from distant regions.
+  >
+  > The app team is instructed to build the smallest container images they can. This is something they could achieve by following the [builder pattern](https://docs.docker.com/develop/develop-images/multistage-build/#before-multi-stage-builds) or the [mutli-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds), both approaches will produce final smaller container images that are meant to be for runtime-only. This will be beneficial in many ways but mainly in the speed of replication as well as in the transfer costs. A great feature as part of the `Geo-Replication` is that it will only replicates unique layers, also reducing data transfer across regions.
```

Comment:
```suggestion
  > The app team is instructed to build the smallest container images they can. This is something they could achieve by following the [builder pattern](https://docs.docker.com/develop/develop-images/multistage-build/#before-multi-stage-builds) or the [mutli-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds). Both approaches will produce final smaller container images that are meant to be for runtime-only. This will be beneficial in many ways but mainly in the speed of replication as well as in the transfer costs. A key feature as part of ACR's geo-replication is that it will only replicate unique layers, also further reducing data transfer across regions.
```